File: NYQ/NYQ/CollectionViewCell.m
//
//  CollectionViewCell.m
//  NYQ
//
//  Created by Zinets Victor on 12/18/15.
//  Copyright © 2015 Zinets Victor. All rights reserved.
//

#import "CollectionViewCell.h"
#import "UIColor+MUIColor.h"

@implementation CollectionViewCell {    
    __weak IBOutlet UIImageView *_imageView;
}

- (void)awakeFromNib {
    self.contentView.backgroundColor = [UIColor colorWithHex:random()];
}

-(void)setImage:(UIImage *)image {
    _imageView.backgroundColor = [UIColor colorWithHex:random()];
}

@end

File: autoCollectionCell/autoCollectionCell/CollectionViewCell.swift
//
//  CollectionViewCell.swift
//  autoCollectionCell
//
//  Created by Victor Zinets on 11/8/18.
//  Copyright © 2018 TN. All rights reserved.
//

import UIKit

class CollectionViewCell: UICollectionViewCell {
    
    @IBOutlet weak var textLabel: UILabel!
    
    override func preferredLayoutAttributesFitting(_ layoutAttributes: UICollectionViewLayoutAttributes) -> UICollectionViewLayoutAttributes {
        setNeedsLayout()
        layoutIfNeeded()
        let size = contentView.systemLayoutSizeFitting(layoutAttributes.size)
        var frame = layoutAttributes.frame
        frame.size = size // width = ceil(size.width)
        layoutAttributes.frame = frame
        return layoutAttributes
    }
}
